### PR TITLE
Malt dbschema

### PR DESCRIPTION
**Pull request by sravfeyn**
better reviewed by commits

### CODE REVIEW

**Comment by sravfeyn** [comment]
@millerdev buddy

@czue FYI, adding some optimizations, as this a good chance before rebuilding the tables again.

**Comment by sravfeyn** [comment]
I tested this using the test I wrote, but the test is not included here because of the problem in https://github.com/dimagi/commcare-hq/pull/7545

**Comment by sravfeyn** [comment]
I didn't write any migrations, as existing MALT table data needs to be deleted anyway, so I am just going to delete the MALT table.

File: corehq/apps/data_analytics/malt_generator.py [hunk 1]
```diff
@@ -62,8 +63,8 @@ def _get_malt_row_dicts(self, user, domain_name, monthspan):
                 'domain_name': domain_name,
                 'num_of_forms': num_of_forms,
                 'app_id': app_id,
-                'wam': wam,
-                'pam': pam,
+                'wam': MALTRow.AMPLIFY_COUCH_TO_SQL_MAP.get(wam, AMPLIFIES_NOT_SET),
+                'pam': MALTRow.AMPLIFY_COUCH_TO_SQL_MAP.get(pam, AMPLIFIES_NOT_SET),
```
Comment (snopoke) [inline corehq/apps/data_analytics/malt_generator.py]: Shouldn't the defaults here be `MALTRow.NOT_SET`?

Comment (sravfeyn) [inline corehq/apps/data_analytics/malt_generator.py]: ha! That's right. Thanks

**Comment by snopoke** [comment]
You should still have an initial migration that creates the table.

**Comment by sravfeyn** [comment]
If I delete the table before this gets deployed, will it still need a migration, since syncdb will create the table again?

**Comment by snopoke** [comment]
No it won't but its better to use migrations than syncdb.

**Comment by czue** [comment]
to @snopoke's point - if you ever needed to add a column later you would then need to add a migration, and an initial migration, and you'd have to tell django/south to fake that first one, and if you just use a migration from the beginning you don't need to worry about any of that.

**Comment by snopoke** [comment]
Thanks for the reasoning @czue

**Comment by sravfeyn** [comment]
ha, thanks! I tried to look into use migrations to delete table data as well and ran into the problem you are describing. I will add initial migration now, and delete the data and table manually anyway, is that good?

**Comment by czue** [comment]
sounds good to me

**Comment by sravfeyn** [comment]
Cool, added initial migration and deleted malt-table on prod/india. This is good to go

**Comment by sravfeyn** [comment]
@czue can you merge this? I will be able to build the table over the weekend.

**Comment by czue** [comment]
looks good

**Comment by sravfeyn** [comment]
Thanks

**Pull request merged**